sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -2638,4 +2638,17 @@ class SQLQuerySuite extends QueryTest with SharedSQLContext {
       }
     }
   }
+
+  test("SPARK-21335: support un-aliased subquery") {
+    withTempView("v") {
+      Seq(1 -> "a").toDF("i", "j").createOrReplaceTempView("v")
+      checkAnswer(sql("SELECT i from (SELECT i FROM v)"), Row(1))
+
+      val e = intercept[AnalysisException](sql("SELECT v.i from (SELECT i FROM v)"))
+      assert(e.message ==
+        "cannot resolve '`v.i`' given input columns: [_auto_generated_subquery_name.i]")
```

Comment:
It's supported since 2.0.X, so definitely there are existing user queires and apps. I'm agreeing with this PR and want to understand the scope of changes. It looks good to me.
```scala
scala> sc.version
res0: String = 2.0.2

scala> Seq(1 -> "a").toDF("i", "j").createOrReplaceTempView("v")

scala> sql("SELECT v.i from (SELECT i FROM v)").show
+---+
|  i|
+---+
|  1|
+---+
```